firstPart/4_winterIsComing.js

```diff
@@ -0,0 +1,19 @@
+const calcPairsOfGloves = (arr) => {
+    const collection = {};
+    arr.forEach(gloveColor => {
+        if (!collection.hasOwnProperty(gloveColor)) {
+            collection[gloveColor] = 0;
+        }
+        collection[gloveColor] += 1;
+    });
+    return Object.values(collection)
+        .map(num => Math.floor(num / 2))
+        .reduce((sum, num) => sum + num, 0);
```

Comment:
the reduce method allows you to store semi result between the iterations. no need to iterates twice. rework your logic to remove the extra map loop

Reply:
Yes, you are right. I've changed it.